### PR TITLE
Check that ip should be different from gateway

### PR DESCRIPTION
**Pull request by AlaaElattar**
### Description

- Complex error when user add ip same as gateway. 
- Wierd validation also happened.
- IPv6 can be added to public config of node without Gateway IPv6

### Changes

- Check they're not the same before sending the request. 
- Fix behaviour of validation.
- Check that both IPv6 and Gateway IPv6 should be added together.
 
### Related Issues

https://github.com/threefoldtech/tfgrid-sdk-ts/issues/1618
https://github.com/threefoldtech/tfgrid-sdk-ts/issues/1668
https://github.com/threefoldtech/tfgrid-sdk-ts/issues/1732

[Screencast from 12-19-2023 01:16:27 PM.webm](https://github.com/threefoldtech/tfgrid-sdk-ts/assets/54944491/a4f044d3-8900-454a-ad0d-1d8db55df786)

### Checklist

- [x] Tests included
- [ ] Build pass
- [ ] Documentation
- [ ] Code format and docstrings
- [x] Screenshots/Video attached (needed for UI changes)

### CODE REVIEW

**Review by zaelgohary — COMMENTED** [review]
I think validation should happen before clicking the button and making the request. If 2 IPs are the same, the button should be disabled.

**Comment by AlaaElattar** [comment]
> I think validation should happen before clicking the button and making the request. If 2 IPs are the same, the button should be disabled.

image

**Review by mohamedamer453 — COMMENTED** [review]
- The validation isn't very accurate in some cases like the one seen below the validation is triggered with incorrect cases and in the case that should trigger it, it doesn't get triggered.

   [Screencast from 13-12-23 13:50:21.webm](https://github.com/threefoldtech/tfgrid-sdk-ts/assets/101194226/74a8ef11-5b23-4591-8206-57cab36c0f0a)

**Comment by AlaaElattar** [comment]
> * The validation isn't very accurate in some cases like the one seen below the validation is triggered with incorrect cases and in the case that should trigger it, it doesn't get triggered.
>   
>     
>       
>       
>   
>       Screencast.from.13-12-23.13.50.21.webm

I think that's because this issue https://github.com/threefoldtech/tfgrid-sdk-ts/issues/1668

**Comment by 0oM4R** [comment]
please check this comment 
https://github.com/threefoldtech/tfgrid-sdk-ts/issues/1668#issuecomment-1860253027

**Review by amiraabouhadid — APPROVED** [review]

File: packages/playground/src/utils/validators.ts [hunk 1]
```diff
@@ -705,6 +705,17 @@ export function isString(msg: string) {
   };
 }
 
+export function ipNotEqualGateway(ip: string, gw: string, msg: string) {
+  return (value: string) => {
+    const ipv = ip.split("/");
+    const ipvAddress = ipv[0];
+
+    if (gw === ipvAddress) {
```
Comment (zaelgohary) [inline packages/playground/src/utils/validators.ts:713]:
```suggestion
  return () => {
    const address = (ip.split("/"))[0];

    if (gw === address) {
```

**Review by zaelgohary — COMMENTED** [review]
All looks good! Just a small enhancement.

File: packages/playground/src/utils/validators.ts (hunk 1 above)
Comment (AlaaElattar) [reply packages/playground/src/utils/validators.ts]: should be fixed

**Review by zaelgohary — APPROVED** [review]

**Pull request merged**